### PR TITLE
chore(server): refactor server Start function [EE-2606]

### PR DESCRIPTION
**Pull request by cheloRydel**
Closes [EE-2606](https://portainer.atlassian.net/browse/EE-2606)

### CODE REVIEW

File: http/server.go [hunk 1]
```diff
@@ -152,26 +112,45 @@ func (server *APIServer) StartSecured(edgeMode bool) error {
...
+		}
+
+		server.edgeManager.ResetActivityTimer()
+
+		next.ServeHTTP(w, r)
+	})
+}
+
+func (server *APIServer) nonEdgeHandler(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```
Comment (chiptus) [inline http/server.go]: isn't this whole function the same as `return next`? why do we need this?

File: http/server.go [hunk 2]
```diff
@@ -88,55 +76,27 @@ func (server *APIServer) StartUnsecured(edgeMode bool) error {
 		EdgeManager:          server.edgeManager,
 		KubeClient:           server.kubeClient,
 		KubernetesDeployer:   server.kubernetesDeployer,
-		Secured:              false,
+		Secured:              !edgeMode,
 		ContainerPlatform:    server.containerPlatform,
 	}
 
-	var h http.Handler = handler.NewHandler(config)
+	var httpHandler http.Handler = handler.NewHandler(config)
```
Comment (chiptus) [inline http/server.go]: I know it was like this, but looks better IMO 
```suggestion
	httpHandler := handler.NewHandler(config)
```

**Review by chiptus — APPROVED** [review]
LGTM. see comments for suggestions

File: http/server.go (hunk 1 above)
Comment (cheloRydel) [reply http/server.go]: yes you're right, I think it can be simplified by just deleting the function

**Review by deviantony — APPROVED** [review]
LGTM

**Pull request merged**